cmd/gctcli/commands.go

```diff
@@ -4604,3 +4049,157 @@ func getCurrencyTradeURL(c *cli.Context) error {
 	jsonOutput(result)
 	return nil
 }
+
+// UnmarshalCLIFields unmarshals field values from command line as *cli.Context to an interface.
+func UnmarshalCLIFields(c *cli.Context, params any) error {
+	val := reflect.ValueOf(params).Elem()
+	typ := val.Type()
+	for i := range typ.NumField() {
+		field := typ.Field(i)
+		flagStrings := strings.Split(field.Tag.Get("cli"), ",")
+		if len(flagStrings) == 0 {
+			continue
+		}
+		required := len(flagStrings) > 1 && strings.EqualFold(flagStrings[len(flagStrings)-1], "required")
+
+		flagNames := flagStrings
+		if len(flagStrings) > 1 && required {
+			flagNames = flagStrings[:len(flagStrings)-1]
+		}
+		switch field.Type.Kind() {
+		case reflect.String:
+			var value string
+			for n := range flagNames {
+				if c.IsSet(flagNames[n]) {
+					value = c.String(flagNames[n])
+				} else {
+					value = c.Args().Get(i)
+				}
+				if value != "" {
+					break
+				}
+			}
+			if required && value == "" {
+				return fmt.Errorf("%w for flag %q", ErrRequiredValueMissing, flagNames[0])
+			}
+			val.Field(i).SetString(value)
+		case reflect.Float64:
+			var value float64
+			for n := range flagNames {
+				if c.IsSet(flagNames[n]) {
+					value = c.Float64(flagNames[n])
+				} else {
+					var err error
+					value, err = strconv.ParseFloat(c.Args().Get(i), 64)
+					if err != nil {
+						return err
+					}
+				}
+				if value != 0 {
+					break
+				}
+			}
+			if required && value == 0 {
+				return fmt.Errorf("%w for flag %q", ErrRequiredValueMissing, flagNames[0])
+			}
+			val.Field(i).SetFloat(value)
+		case reflect.Bool:
+			var value bool
+			for n := range flagNames {
+				if c.IsSet(flagNames[n]) {
+					value = c.Bool(flagNames[n])
+				} else {
+					var err error
+					value, err = strconv.ParseBool(c.Args().Get(i))
+					if required && (err != nil || !value) {
+						return fmt.Errorf("%w for flag %q", ErrRequiredValueMissing, flagNames[0])
+					}
+				}
+				if !value {
+					break
+				}
+			}
+			val.Field(i).SetBool(value)
+		case reflect.Int64:
+			var value int64
+			for n := range flagNames {
+				if c.IsSet(flagNames[n]) {
+					value = c.Int64(flagNames[n])
+				} else {
+					var err error
+					value, err = strconv.ParseInt(c.Args().Get(i), 10, 64)
+					if err != nil {
+						return err
+					}
+				}
+				if value != 0 {
+					break
+				}
+			}
+			if required && value == 0 {
+				return fmt.Errorf("%w for flag %q", ErrRequiredValueMissing, flagNames[0])
+			}
+			val.Field(i).SetInt(value)
+		}
+	}
+	return nil
+}
+
+// FlagsFromStruct returns list of cli flags from exported flags
+func FlagsFromStruct(params any, usageInfo map[string]string) []cli.Flag {
```

Comment:
suggestion: can use specific tags for  `field.Tag.Get` as `name,usage, required` for field and set destination for value, if no usage can set up a global map for default values if no name default to using the field name and try to standardise field names.